### PR TITLE
Replace keycloak-proxy by oauth2-proxy

### PR DESCRIPTION
**Pull request by chrkl**
**Description**

Changes proposed in this pull request:

- Replace keycloak-proxy by oauth2 proxy
- Make proxy settings configurable by user

**Related issue(s)**

See also #11181

### CODE REVIEW

File: resources/kiali/templates/kyma-additions/auth-proxy-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  OAUTH2_PROXY_CLIENT_ID: {{ .Values.authProxy.config.clientId | b64enc | quote }}
+  OAUTH2_PROXY_CLIENT_SECRET: {{ .Values.authProxy.config.clientSecret | b64enc | quote }}
+  OAUTH2_PROXY_EMAIL_DOMAINS: {{ .Values.authProxy.config.emailDomains | b64enc | quote }}
+  OAUTH2_PROXY_COOKIE_SECRET: {{ randAlphaNum 32 | b64enc | quote }}
+  {{- if .Values.authProxy.configDocsLink }}
+  OAUTH2_PROXY_CUSTOM_TEMPLATES_DIR: {{ "/templates" | b64enc | quote }}
+  {{- end }}
+  OAUTH2_PROXY_SSL_INSECURE_SKIP_VERIFY: {{ not .Values.authProxy.config.tlsVerify | toString | b64enc | quote }}
+  {{- if .Values.authProxy.config.useKymaGroups }}
+  OAUTH2_PROXY_ALLOWED_GROUPS: {{ template "kiali.authProxy.kymaGroups" . | b64enc | quote }}
```
Comment (suleymanakbas91) [inline resources/kiali/templates/kyma-additions/auth-proxy-secret.yaml:19]:
```suggestion
  OAUTH2_PROXY_ALLOWED_GROUPS: {{ template "kiali.kyma.authProxy.kymaGroups" . | b64enc | quote }}
```

File: resources/tracing/templates/_helpers.tpl [hunk 2]
```diff
@@ -59,24 +59,8 @@ Create chart name and version as used by the chart label.
...
-{{- if .Values.kcproxy.config.resources.methods }}
-{{- printf "|methods=%s" .Values.kcproxy.config.resources.methods }}
-{{- end }}
-{{- end -}}
-
-{{- define "jaeger-operator.kcproxy.roles" -}}
-{{- if .Values.kcproxy.config.resources.roles }}
-{{- printf "|roles=%s" .Values.kcproxy.config.resources.roles }}
-{{- end }}
+{{- define "jaeger-operator.authProxy.kymaGroups" -}}
```
Comment (suleymanakbas91) [inline resources/tracing/templates/_helpers.tpl:62]: Let's make it consistent with the others :) 
```suggestion
{{- define "jaeger-operator.kyma.authProxy.kymaGroups" -}}
```

File: resources/tracing/templates/kyma-additions/auth-proxy-secret.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  OAUTH2_PROXY_CLIENT_ID: {{ .Values.authProxy.config.clientId | b64enc | quote }}
+  OAUTH2_PROXY_CLIENT_SECRET: {{ .Values.authProxy.config.clientSecret | b64enc | quote }}
+  OAUTH2_PROXY_EMAIL_DOMAINS: {{ .Values.authProxy.config.emailDomains | b64enc | quote }}
+  OAUTH2_PROXY_COOKIE_SECRET: {{ randAlphaNum 32 | b64enc | quote }}
+{{- if .Values.authProxy.configDocsLink }}
+  OAUTH2_PROXY_CUSTOM_TEMPLATES_DIR: {{ "/templates" | b64enc | quote }}
+{{- end }}
+  OAUTH2_PROXY_SSL_INSECURE_SKIP_VERIFY: {{ not .Values.authProxy.config.tlsVerify | toString | b64enc | quote }}
+{{- if .Values.authProxy.config.useKymaGroups }}
+  OAUTH2_PROXY_ALLOWED_GROUPS: {{ template "jaeger-operator.authProxy.kymaGroups" . | b64enc | quote }}
```
Comment (suleymanakbas91) [inline resources/tracing/templates/kyma-additions/auth-proxy-secret.yaml:18]:
```suggestion
  OAUTH2_PROXY_ALLOWED_GROUPS: {{ template "jaeger-operator.kyma.authProxy.kymaGroups" . | b64enc | quote }}
```

**Review by suleymanakbas91 — CHANGES_REQUESTED** [review]

File: resources/kiali/templates/kyma-additions/auth-proxy-secret.yaml (hunk 1 above)
Comment (chrkl) [reply resources/kiali/templates/kyma-additions/auth-proxy-secret.yaml]: Done

File: resources/tracing/templates/_helpers.tpl (hunk 2 above)
Comment (chrkl) [reply resources/tracing/templates/_helpers.tpl]: Done

File: resources/tracing/templates/kyma-additions/auth-proxy-secret.yaml (hunk 3 above)
Comment (chrkl) [reply resources/tracing/templates/kyma-additions/auth-proxy-secret.yaml]: Done

**Review by suleymanakbas91 — APPROVED** [review]

**Review by suleymanakbas91 — DISMISSED** [review]

**Review by crabtree — APPROVED** [review]

**Review by mjakobczyk — APPROVED** [review]

**Review by crabtree — APPROVED** [review]

**Review by suleymanakbas91 — APPROVED** [review]

**Pull request merged**